xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiAction.java

```diff
@@ -1221,4 +1158,27 @@ protected boolean readFromTemplate(XWikiDocument document, String template, XWik
 
         return false;
     }
+
+    /**
+     * Loop over the {@link RedirectionFilter} components until one of them perform a redirection. If none of the does,
+     * the action continues normally.
+     *
+     * @param context the current wiki content
+     * @return {@code true} if a redirection has been performed, {@code false} otherwise
+     * @throws XWikiException in case of error during the execution of a redirection filter
+     */
+    private boolean handleRedirect(XWikiContext context) throws XWikiException
+    {
+        // If no redirection are expected, this step is skipped.
+        if (this.handleRedirectObject) {
+            List<RedirectionFilter> filters = Utils.getComponentList(RedirectionFilter.class);
```

Comment:
XWikiActions are component since 13.0 so not need to go through Utils.getComponent*.

Reply:
Replaced by `this.componentManager.<RedirectionFilter>getInstanceList(RedirectionFilter.class)`.